### PR TITLE
Make ruby RpcServer stop waiting for calls when it gets null calls

### PR DESCRIPTION
**Pull request by murgatroid99**
In the lines changed, a `nil` call implies that the server has shut down, so waiting for further calls is guaranteed to be fruitless. This was trying to help with #3598, but further testing indicates that this was probably not the root cause there. It should at least substantially shorten error logs, though.

This also makes RuboCop a little more lenient, so that I could actually make this change.

### CODE REVIEW

File: src/ruby/.rubocop.yml [hunk 1]
```diff
@@ -9,3 +9,9 @@ AllCops:
     - 'bin/math_services.rb'
     - 'pb/grpc/health/v1alpha/*'
     - 'pb/test/**/*'
+
+Metrics/CyclomaticComplexity:
+  Max: 8
+
+Metrics/PerceivedComplexity:
+  Max: 8
```
Comment (jtattermusch) [inline src/ruby/.rubocop.yml]: nit: add EOL

**Comment by jtattermusch** [comment]
LGTM and fine to merge once green.

**Comment by jtattermusch** [comment]
It looks like that after this change got merged, all tests against ruby server started timing out.

**Comment by murgatroid99** [comment]
I'm not sure what you're seeing. As far as I can tell, ruby tests haven't failed on Jenkins master or pull requests since this was merged.

**Comment by jtattermusch** [comment]
https://grpc-testing.appspot.com/job/gRPC_interop_master/305/#showFailuresLink  but it might me just flakiness.

**Pull request merged**